### PR TITLE
Add another round of tests.

### PR DESCRIPTION
**Pull request by JayNewstrom**

### CODE REVIEW

**Review by arvindkrishnakumar-okta — APPROVED** [review]

File: app/src/androidTest/java/com/okta/idx/android/SelfServicePasswordRecoveryTest.kt [hunk 1]
```diff
@@ -47,23 +43,17 @@ class SelfServicePasswordRecoveryTest {
         private const val ERROR_TEXT_VIEW = "com.okta.idx.android:id/error_text_view"
         private const val ID_TOKEN_TYPE_TEXT_VIEW = "com.okta.idx.android:id/token_type"
         private const val CODE_EDIT_TEXT = "com.okta.idx.android:id/code_edit_text"
-        private const val CONFIRMED_PASSWORD_EDIT_TEXT = "com.okta.idx.android:id/confirmed_password_edit_text"
+        private const val CONFIRMED_PASSWORD_EDIT_TEXT =
```
Comment (NikitaAvraimov-okta) [inline app/src/androidTest/java/com/okta/idx/android/SelfServicePasswordRecoveryTest.kt:46]: We have same ids in some layouts e.g. username_text_input_layout in form_forgot_password and form_username_and_password, wouldn`t that be a problem?

Comment (JayNewstrom) [reply app/src/androidTest/java/com/okta/idx/android/SelfServicePasswordRecoveryTest.kt:46]: Not a problem per se. But would be better to consolidate them. I'll do it in a follow up.

**Comment by NikitaAvraimov-okta** [comment]
Will we leave mocked tests along with e2e or will they be removed?

**Comment by JayNewstrom** [comment]
I'd like to have both tests (mocks, e2e) running side by side long term. Short term, copy/paste to hit deadlines.

e2e are great for us to catch regressions, but hard for external developers. Mocks are easy for external developers to run tests and validate their change doesn't break anything. That way they don't have to run on our internal infrastructure to make sure they didn't break anything.

**Pull request merged**